bitcoin/src/psbt/mod.rs

```diff
@@ -731,6 +728,87 @@ impl Psbt {
     }
 }
 
+#[cfg(feature = "serde")]
+impl serde::Serialize for Psbt {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer {
+
+        let bytes = self.serialize();
+
+        if serializer.is_human_readable() {
+            use base64::Engine;
+
+            let s = base64::prelude::BASE64_STANDARD.encode(bytes);
+
+            serializer.serialize_str(s.as_ref())
+        } else {
+            serializer.serialize_bytes(bytes.as_ref())
+        }
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<'de> serde::Deserialize<'de> for Psbt {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>
+    {
+        use core::marker::PhantomData;
+
+        struct Visitor(PhantomData<Psbt>);
```

Comment:
I'm no `sered` expert but don't think we need this `PhantomData`. Did you copy this code from `amount`? If so the usecase is different.

Reply:
Ugh you're totally right. I think I copied it from one of the (now removed) implementations in `bitcoin/src/serde_utils.rs` FWIW.